docs/datatypes/matrices.md

```diff
@@ -331,6 +331,44 @@ method `.set()`, the matrix will be resized. By default, new items will be
 initialized with zero, but it is possible to specify an alternative value using
 the optional third argument `defaultValue`.
 
+## Advanced Indexing
+
+It is also possible to use an array of booleans for index. This allows some advanced 
+capabilities as the array can come from conditions of other functions it allows for 
+filtering, replacing many values and setting specific values.
```

Comment:
The sentence "This allows some .... setting specific values." is run-on. Please rewrite and/or split in two, thanks.